Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1273,6 +1273,12 @@ void FrameLoader::loadInSameDocument(URL url, RefPtr<SerializedScriptValue> stat
 
     checkedHistory()->updateForSameDocumentNavigation();
 
+    RefPtr domWindow = m_frame->document()->protectedWindow();
+    if (domWindow && m_frame->document()->settings().navigationAPIEnabled()) {
+        domWindow->protectedNavigation()->dispatchNavigationEvent(url, oldURL, documentLoader()->triggeringAction().type());
```

Comment:
Will this affect non-users of the new API when the new API is turned on?

Reply:
I can't really foresee any reason it would.

Any state it touches is contained to the Navigation object.